### DISCUSSION TITLE
Do I need to know rust as a frontend developer?

### DISCUSSION THREAD

**Discussion by arashi-dev** [tauri-apps/tauri#3343] [Q&A] [answered]
Hello.
I am interested in `Tauri` framework, and I only know javascript (typescript).
so I have some questions about it

1. Do I need to know `rust` to be able to build an app?
2. which parts of a `Tauri` app requires rust code?
3. how much knowledge do I need, to use rust for Tauri, anyway?
4. can I use NodeJS for backend? (e.g. external API)

**Comment by mrceperka** [comment, marked as answer]
IMHO:
1. no
2. "backend" or "wiring"
3. almost none, `yarn tauri build` handlers that for you
4. as remote api yes, you can. as "local" it's harder but possible with sidecar.

**Comment by yeasin2002** [comment]
so, how  can i use native desktop  features ?

**Reply by arashi-dev** [reply]
Hey. I just started using Tauri a few weeks ago.

You use Rust. The core of a tauri application will be executed from rust. Rust tells the OS to open our HTML files in the Webview. Tauri SDK also lets you invoke the functions written inside the rust files, therefore you can use the native APIs using rust and then invoke the functions inside JS, which returns the promise of the result.

However, the amazing part is that Tauri includes some built-in APIs that let you access some of the most important native APIs inside JS without any issues. more information in the document: https://tauri.app/v1/api/js/

PS. There are some plugins (some official and some from the community) that are not built-in inside Tauri but can be added easily. You can find them here: https://github.com/tauri-apps/awesome-tauri?tab=readme-ov-file#plugins

PS 2. You can also use the sidecars but I didn't test them yet. By sidecars, you can embed an external binary (e.g. NodeJS, Python, Deno, etc.) in your application. more information: https://tauri.app/v1/guides/building/sidecar/

PS 3. (Just an experience sharing) It's obvious and we've read many times that Rust is a difficult language but my survey of learning it in Tauri was/is really interesting! The documentation and the ease of the Tauri APIs are really amazing. You may just need to watch a mini-tutorial of rust on YouTube (4~5 hours are enough to go)
